hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -27,38 +31,53 @@ export default function HackathonSelectDropdown(setHackathonID: Props) {
     )
   }
 
-  const hackathonMap = hackathonList.map((hackathon, index) => ({
+  const hackathonDateCurrentAndFuture = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) > today
+  })
+
+  const hackathonDatePast = hackathonList.filter((hackathon) => {
+    return new Date(hackathon.endDate) < today
+  })
+
+  const hackathonMap = (
+    context === 'archive'
+      ? hackathonDatePast
+      : context === 'idea-portal'
+      ? hackathonDateCurrentAndFuture
+      : hackathonList
```

Comment:
This is hard to read, wdyt about:
```suggestion
const hackathonMap = getHackatonsForContext()
```

```
function getHackatonsForContext(): HackathonPreview[] {
    switch(context){
      case 'archive':
        return hackathonDatePast;
      case 'ida-portal':
        return futureAndPresentHackatons;
    }
    return hackathonList;
  }
```